src/System.Windows.Extensions/src/System/Media/SoundPlayer.cs

```diff
@@ -0,0 +1,738 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Runtime.InteropServices;
+using System.Runtime.Serialization;
+using System.Threading;
+
+namespace System.Media
+{
+    [Serializable]
+    [ToolboxItem(false)]
+    public class SoundPlayer : Component, ISerializable
+    {
+        private const int BlockSize = 1024;
+        private const int DefaultLoadTimeout = 10000; // 10 secs
+
+        private Uri _uri = null;
+        private string _soundLocation = string.Empty;
+        private int _loadTimeout = DefaultLoadTimeout;
+
+        // used to lock all synchronous calls to the SoundPlayer object
+        private ManualResetEvent _semaphore = new ManualResetEvent(true);
```

Comment:
Nit: readonly.  I'm also not sure why this is using an MRE as a semaphore, but that could be looked at separately... better not to mess with any synchronization as part of this PR.